tensorflow/python/framework/ops.py

```diff
@@ -372,7 +384,7 @@ def __init__(self, op, value_index, dtype):
       raise TypeError("op needs to be an Operation: %s" % op)
     self._op = op
     self._value_index = value_index
-    self._dtype = dtypes.as_dtype(dtype)
+    self._dtype = dtypes.as_dtype(dtype) # type: DataType
```

Comment:
Why is this type annotation needed? `as_dtype` is used in many places so it'd be nice to get the annotations all over the place.